tests/configurations/integration/jest.globalSetup.ts

```diff
@@ -1,9 +1,47 @@
+/* eslint-disable no-console */
+import path from 'node:path';
+import * as fs from 'fs';
+import * as compose from 'docker-compose';
 import { $ } from 'zx';
 import type { commonDbFullV1Type } from '@map-colonies/schemas';
+import { getPort, checkPort } from 'get-port-please';
 import { initConfig, getConfig } from '../../../src/common/config';
 import { createDbConnectUrl } from '../../../src/db/helpers';
 
+const configPath = path.join(process.cwd(), 'config', 'local-test.json');
+interface Config {
+  db?: {
+    port?: number;
+  };
+}
+
+async function getSelectedPort(config: Config): Promise<number> {
+  let port: number;
+
+  if (config.db?.port !== undefined) {
+    port = config.db.port;
+    console.log(`Found custom port in local-test.json. Using port: ${port}`);
+  } else {
+    // If a port was not defined in the config file, find an available one and write it to local-test.json
+    port = await getPort({ port: 5433 });
+    console.log(`No custom port defined. Using available port: ${port}`);
+    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+    fs.writeFileSync(configPath, JSON.stringify({ ...config, db: { ...config.db, port } }, null, 2), 'utf-8');
```

Comment:
you provide the config as input, but save the config again inside the function

Reply:
Separated